build.assets/notarize-binaries.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
```

Comment:
I might be a bit late to the party here, but is it worth translating this to Go? For legibility we've generally tried to move anything vaguely complex from shell scripts to Go.

Reply:
@fheinecke I agree, is it a lot of work to write a Go program that does this? This will help us avoid scary looking things like "shell_array_to_json_array" below.

I'd be ok to merge this and replace this with Go program as a follow up right after.

Reply:
The primary reason I didn't write this in Go is because all the other notarization code is written as a bash script (`build.assets/build-common.sh`, `build.assets/build-package.sh`). Originally I thought this was going to be a much simpler task that would require me just calling a couple of functions in either `build-common.sh` or calling `build-package.sh` with some changes. However as I worked on the ticket I found that neither would work without significant changes to other areas (due to coupling), so instead I copied a lot of my work from those files and modified it slightly to fit my needs.

In my opinion rewriting this in Go would add a bit of complexity to our build pipelines, as notarizing `pkg`s would still be done with shell scripts. That being said I think it's probably worth the tradeoff assuming we want to allocate the time to do so.

Reply:
+1 for writing in Go. I would have a hard time changing this script and being confident that I didn't break anything.

Reply:
Rewrote the tool in go, marking this as resolved